File: backend/src/test/java/com/NBE_4_5_2/Team5/api/TestErrorController.java
package com.NBE_4_5_2.Team5.api;

import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/test")
public class TestErrorController {

    @GetMapping("/bad-request")
    public void triggerBadRequest() {
        throw new IllegalArgumentException();
    }

    @GetMapping("/nullPointer-error")
    public void triggerNullPointerException() {
        throw new NullPointerException();
    }
}

File: backend/src/test/java/com/NBE_4_5_2/Team5/api/TestErrorController.kt
package com.NBE_4_5_2.Team5.api

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/test")
class TestErrorController {
    @GetMapping("/bad-request")
    fun triggerBadRequest() {
        throw IllegalArgumentException()
    }

    @GetMapping("/nullPointer-error")
    fun triggerNullPointerException() {
        throw NullPointerException()
    }
}
